mmf/trainers/core/reporting.py

```diff
@@ -37,7 +37,10 @@ def update_meter(
 
                 registry.register(loss_key, reduced_loss)
                 meter_update_dict.update({loss_key: reduced_loss})
-                meter_update_dict.update(reduced_loss_dict)
+                for key in reduced_loss_dict:
+                    loss = reduced_loss_dict[key].item()
+                    meter_update_dict[key] = loss
+
             if hasattr(report, "metrics"):
                 meter_update_dict.update(reduced_metrics_dict)
```

Comment:
We should this for metrics as well if metric hasattr item to be foolproof? In rare cases, this may also blow out when metric actually doesn't do .item itself.

Reply:
> We should this for metrics as well if metric hasattr item to be foolproof? In rare cases, this may also blow out when metric actually doesn't do .item itself.

Ok, in the latest version I refactored updating `reduced_metrics_dict` and `reduced_loss_dict` to a single update function which checks for `.item()`.